src/usermod.c

```diff
@@ -2039,8 +2046,8 @@ static void move_mailbox (void)
 	if (NULL == maildir) {
 		return;
 	}
-	len = strlen (prefix) + strlen (maildir) + strlen (user_name) + 2;
-	mailfile = ALLOCARRAY (len, char);
+	size = strlen (prefix) + strlen (maildir) + strlen (user_name) + 2;
+	mailfile = XMALLOCARRAY (size, char);
```

Comment:
Isn't this an off-by-one bug?  I think this may be what confused me.  We add 2 bytes, but we need 3 for the two `/` and the terminating NUL:

```c
	(void) snprintf (mailfile, size, "%s/%s/%s", prefix, maildir, user_name);
```

Reply:
Indeed, nice catch!